ykrt/src/trace/sw/mod.rs

```diff
@@ -0,0 +1,118 @@
+//! Software tracer.
+
+use crate::frame::BitcodeSection;
+
+use super::{errors::InvalidTraceError, TraceCollector, TraceIterator, TracedAOTBlock};
+use std::{cell::RefCell, collections::HashMap, error::Error, ffi::CString, sync::Arc};
+
+#[derive(Debug, Eq, PartialEq)]
+struct BB {
+    function_index: u32,
+    block_index: u32,
+}
+
+thread_local! {
+    static BASIC_BLOCKS: RefCell<Vec<BB>> = RefCell::new(vec![]);
+}
+
+extern "C" {
+    fn get_function_names(
+        section: *const BitcodeSection,
+        vec: *const libc::c_uint,
+        vec_size: usize,
+        result: *mut *mut IRFunctionNameIndex,
+        len: *mut libc::size_t,
+    );
+}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct IRFunctionNameIndex {
+    pub index: u32, //libc::c_int,
```

Comment:
Commented code. Also should this be `usize`?

Reply:
comment removed 👉 2690d9c5c195c0175655c32e4e342e7871701e56

Reply:
We can use `usize` here, but at some point, we'll have to cast it to `u32` anyway cause that's the type for indices coming from the LLVM pass.

Reply:
> or maybe all the indices should be usize anyway?

In general I'd start with `usize` unless you know that the space is definitely a problem.

Reply:
changed to `usize` 👉 7f2664dc28755524ed4461d141f2478d26c274ec